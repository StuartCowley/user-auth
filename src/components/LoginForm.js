import React, {useState} from "react";

import "../styles/login-form.css";

const LoginForm = () => {
  const initialFormValues = {
    email: "",
    password: ""
  };
  const [formValues, setFormValues] = useState(initialFormValues);

  const handleInputChange = (e) => {
    const updatedValues = { ...formValues };
    updatedValues[e.target.id] = e.target.value
    
    setFormValues(updatedValues);
  };

  const handleSubmit = () => {
    // TODO implement login
  }

  return (
    <div className="login-form">
      <h1>Login here:</h1>
      <div className="login-form__element">
        <label htmlFor="email">Email</label>
        <input
          className="login-form__input"
          type="email"
          id="email"
          onChange={handleInputChange} 
          value={formValues.email} 
        />
      </div>
      <div className="login-form__element">
        <label htmlFor="password">Password</label>
        <input
          className="login-form__input"
          type="password"
          id="password"
          onChange={handleInputChange} 
          value={formValues.password} 
        />
      </div>
      <button onClick={handleSubmit}>Submit</button>
    </div>
  )
}

export default LoginForm