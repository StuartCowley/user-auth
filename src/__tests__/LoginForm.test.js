import { render } from '@testing-library/react';
import renderer from "react-test-renderer";
import LoginForm from '../components/LoginForm';

describe("Login form component", () => {
  it("matches snapshot", () => {
    const generated = renderer
    .create(<LoginForm />)
    .toJSON();

  expect(generated).toMatchSnapshot();
  })
  
  it("renders title", () => {
    const {getByText} = render(<LoginForm />);
    const title = getByText(/Login here:/);

    expect(title).toBeInTheDocument;
  })
});