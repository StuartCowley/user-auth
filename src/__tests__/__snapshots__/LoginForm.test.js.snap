// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login form component matches snapshot 1`] = `
<div
  className="login-form"
>
  <h1>
    Login here:
  </h1>
  <div
    className="login-form__element"
  >
    <label
      htmlFor="email"
    >
      Email
    </label>
    <input
      className="login-form__input"
      id="email"
      onChange={[Function]}
      type="email"
      value=""
    />
  </div>
  <div
    className="login-form__element"
  >
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      className="login-form__input"
      id="password"
      onChange={[Function]}
      type="password"
      value=""
    />
  </div>
  <button
    onClick={[Function]}
  >
    Submit
  </button>
</div>
`;
